jbi/runner.py

```diff
@@ -157,6 +158,29 @@ def __call__(self, context: ActionContext) -> ActionResult:
         return True, {"responses": responses}
 
 
+async def execute_or_queue(
+    request: bugzilla.WebhookRequest, queue: DeadLetterQueue, actions: Actions
+):
+    if await queue.is_blocked(request):
```

Comment:
I think we should think about gracefully handling when the queue is down, somehow. 

For short-term outages, I think it's enough to retry checking if the queue is blocked and/or postponing the bug until we can connect again. But for long-term outages, I'm not sure what we can do 🤔.

For example, say a bug has 2 events in the queue, but also the queue is unavailable for some reason. What should we do with a newly incoming webhook payload for the queued bug? 

JBI can't know about the bugs in the queue if it can't connect to it. So does it hold on to bugs until the queue becomes available and it can process them again? Does it go "even though I might have queued bugs for this incoming webhook, I'm going to try processing them anyway"?

Reply:
I may not see every aspect of this, but I think this is good enough to return a 5XX if the queue fails.

If we return a 5XX we are in the same situation as we are now when we fail at processing the event: bugzilla will retry. And eventually disable the hook if fails for a long time, but that would be a service disruption, hence consistent.

Reply:
Adjusting so that if the queue has an error we return 500. Adding unit test to verify.